src/component/solenoid.go

```diff
@@ -0,0 +1,26 @@
+package component
+
+// Solenoid - base component + solenoid specific metadata
+type Solenoid struct {
+	BaseComponent
+	Voltage int
+	Type    SolenoidType
+}
+
+func (s *Solenoid) open(duration int) {
+
+}
+
+func (s *Solenoid) close(delay int) {
```

Comment:
Same as above

Reply:
WIP party, OnState is likely going to be replaced by an RPIO pin object